Microsoft.Toolkit.Uwp.UI/AdvancedCollectionViewSource/AdvancedCollectionViewSource.cs

```diff
@@ -0,0 +1,592 @@
+        /// <returns>success of operation</returns>
+        public bool MoveCurrentToLast() => MoveCurrentToIndex(_view.Count - 1);
+
+        /// <summary>
+        /// Move current item to next item
+        /// </summary>
+        /// <returns>success of operation</returns>
+        public bool MoveCurrentToNext() => MoveCurrentToIndex(_index + 1);
+
+        /// <summary>
+        /// Move current item to previous item
+        /// </summary>
+        /// <returns>success of operation</returns>
+        public bool MoveCurrentToPrevious() => MoveCurrentToIndex(_index - 1);
+
+        /// <summary>
+        /// Load more items from the source
+        /// </summary>
+        /// <param name="count">number of items to load</param>
+        /// <returns>Async operation of LoadMoreItemsResult</returns>
+        /// <exception cref="NotImplementedException">Not implemented yet...</exception>
+        public IAsyncOperation<LoadMoreItemsResult> LoadMoreItemsAsync(uint count)
+        {
+            var isil = _source as ISupportIncrementalLoading;
+            return isil?.LoadMoreItemsAsync(count);
+        }
+
+        /// <summary>
+        /// Gets the groups in collection
+        /// </summary>
+        public IObservableVector<object> CollectionGroups => null;
+
+        /// <summary>
+        /// Gets or sets the current item
+        /// </summary>
+        public object CurrentItem
+        {
+            get { return _index > -1 && _index < _view.Count ? _view[_index] : null; }
+            set { MoveCurrentTo(value); }
+        }
+
+        /// <summary>
+        /// Gets the position of current item
+        /// </summary>
+        public int CurrentPosition => _index;
+
+        /// <summary>
+        /// Gets a value indicating whether the source has more items
+        /// </summary>
+        public bool HasMoreItems => (_source as ISupportIncrementalLoading)?.HasMoreItems ?? false;
+
+        /// <summary>
+        /// Gets a value indicating whether the current item is after the last visible item
+        /// </summary>
+        public bool IsCurrentAfterLast => _index >= _view.Count;
+
+        /// <summary>
+        /// Gets a value indicating whether the current item is before the first visible item
+        /// </summary>
+        public bool IsCurrentBeforeFirst => _index < 0;
+
+        /// <summary>
+        /// Current item changed event handler
+        /// </summary>
+        public event EventHandler<object> CurrentChanged;
+
+        /// <summary>
+        /// Current item changing event handler
+        /// </summary>
+        public event CurrentChangingEventHandler CurrentChanging;
+
+        /// <summary>
+        /// Gets a value indicating whether this CollectionView can filter its items
+        /// </summary>
+        public bool CanFilter => true;
+
+        /// <summary>
+        /// Gets or sets the predicate used to filter the visisble items
+        /// </summary>
+        public Predicate<object> Filter
```

Comment:
I'd personally go with the functional-ish/linq-ish style of the predicate, rather than have events instead. Also we can always ease-in breaking changes by marking things obsolete first. But I guess this is open for discussion.